src/fondant/pipeline/lightweight_component.py

```diff
@@ -281,7 +313,21 @@ def build_python_script(component_cls: t.Type[Component]) -> str:
     """,
     )
 
-    component_source = inspect.getsource(component_cls)
+    if is_running_in_notebook():
+        from IPython.core.magics.code import extract_symbols
+
+        inspect.getfile = new_getfile
+        component_source = "".join(
+            inspect.linecache.getlines(  # type: ignore[attr-defined]
+                new_getfile(component_cls),
+            ),
+        )
+        component_source = extract_symbols(component_source, component_cls.__name__)[0][
+            0
+        ]
```

Comment:
Y u do this Black :cry: 

```suggestion
        component_source = extract_symbols(
            component_source, 
            component_cls.__name__,
        )[0][0]
```

Can you try changing it to this and see if Black keeps it this way? :stuck_out_tongue:

Reply:
haha it does this
```python
  component_source = extract_symbols(
      component_source,
      component_cls.__name__,
  )[
      0
  ][0]
```

Reply:
i'll break it into two lines